libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -540,6 +540,16 @@ bool AP_BattMonitor::capacity_remaining_pct(uint8_t &percentage, uint8_t instanc
     return false;
 }
 
+/// time_remaining - returns remaining battery time
+bool AP_BattMonitor::time_remaining(uint32_t &seconds, uint8_t instance) const
+{
+    if (instance < _num_instances && drivers[instance] != nullptr && drivers[instance]->has_time_remaining()) {
+        seconds = state[instance].time_remaining;
```

Comment:
It's a bit off asking the driver if it has time remaining and then fetching that from state.  Considering these states are often updated with locks held, it's a bit suspicious.

should we move `         uint32_t    time_remaining;            // remaining battery time` into the state from the driver?